integrated_channels/integrated_channel/models.py

```diff
@@ -205,6 +206,76 @@ def clean(self):
                 }
             )
 
+    def get_learner_data_audit_model(self):
+        return LearnerDataTransmissionAudit.get_completion_class_by_channel_code(self.channel_code())
+
+    def get_last_learner(self, error, config_id):
+        """
+        Gets every learner audit associated with the configuration and finds the most recent record.
+        If the boolean parameter 'error' is True, then we are only looking for the most recent
+        audit record where the status > 400.
+        """
+        audit_model = self.get_learner_data_audit_model()
+        if error:
+            learner_audits = audit_model.objects.filter(
+                enterprise_customer_uuid=self.enterprise_customer.uuid,
+                plugin_configuration_id=config_id,
+                status__gte=400,
+            )
+        else:
+            learner_audits = LearnerDataTransmissionAudit.objects.filter(
+                enterprise_customer_uuid=self.enterprise_customer.uuid,
+                plugin_configuration_id=config_id,
+            )
+        if learner_audits:
+            return learner_audits.aggregate(max=Max('completed_timestamp')).get('max')
+        else:
+            return None
+
+    def get_last_content(self, error, config_id):
+        """
+        Gets every content transmission associated with the configuration and finds the most
+        recent record. If the boolean parameter 'error' is True, then we are only looking for
+        the most recent transmission record where the status > 400.
+        """
+        if error:
+            content_audits = ContentMetadataItemTransmission.objects.filter(
+                enterprise_customer=self.enterprise_customer,
+                plugin_configuration_id=config_id,
+                integrated_channel_code=self.channel_code(),
+                api_response_status_code__gte=400,
+            )
+        else:
+            content_audits = ContentMetadataItemTransmission.objects.filter(
+                enterprise_customer=self.enterprise_customer,
+                plugin_configuration_id=config_id,
+                integrated_channel_code=self.channel_code(),
+            )
+        if content_audits:
+            max_created = (content_audits.aggregate(max_created=Max('remote_created_at'))).get('max_created')
+            max_updated = (content_audits.aggregate(max_updated=Max('remote_updated_at'))).get('max_updated')
+            max_deleted = (content_audits.aggregate(max_deleted=Max('remote_deleted_at'))).get('max_deleted')
+            return max(x for x in [max_created, max_updated, max_deleted] if x is not None)
```

Comment:
you may need to check to see if they're all None and return None in that case because I believe `max()` will error if the array ends up being all `None`


```
>>> max(x for x in [None, None, None] if x is not None)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
ValueError: max() arg is an empty sequence
```